light/registry/models/starspace_model.py

```diff
@@ -41,24 +42,29 @@ def get_model(self, overrides: Optional[Dict[str, Any]] = None) -> Agent:
             opt = deepcopy(use_shared["opt"])
             opt.update(overrides)
             use_shared["opt"] = opt
```

Comment:
If I am not mistaken, this modifies the `self._shared` as well (because of `use_shared` reference). Is that intended?

Reply:
Correct, we keep the most recent 'opt' sett.